src/main/java/com/google/api/codegen/viewmodel/FileHeaderView.java

```diff
@@ -40,6 +40,12 @@ public boolean hasVersion() {
     return version() != null && version().length() > 0;
   }
 
+  public abstract String generatorVersion();
+
+  public boolean hasGeneratorVersion() {
+    return generatorVersion() != null && generatorVersion().length() > 0;
```

Comment:
I wasn't asking for anything that complicated. All I meant was:

```java
public boolean hasGeneratorVersion() {
    String version = generatorVersion();
    return version != null && version.length() > 0;
}
```

Sorry for the ambiguity.